flink-table/flink-table-planner/src/test/java/org/apache/flink/table/planner/functions/BuiltInFunctionTestBase.java

```diff
@@ -129,6 +129,8 @@ private static void testTableApiError(Table inputTable, TableApiErrorTestItem te
         try {
             inputTable.select(testItem.expression).execute();
             fail("Error expected: " + testItem.errorMessage);
+        } catch (AssertionError e) {
```

Comment:
(If the `fail` call is reached, this will propagate that assertion failure rather than running into the `catch` and then failing due to the assertion on the (what is supposed to be the expected) exception)